pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,18 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
-    String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
-    if (timeout != null) {
+    long result = 0L;
+    if (PGProperty.LOGIN_TIMEOUT.isPresent(props)) {
+      String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = (long) (Float.parseFloat(timeout) * 1000);
       } catch (NumberFormatException e) {
         LOGGER.log(Level.WARNING, "Couldn't parse loginTimeout value: {0}", timeout);
       }
+    } else {
+      result = (long) DriverManager.getLoginTimeout() * 1000;
```

Comment:
Hi Dave, sounds like a good idea.

I do have one suggestion though. If we go with throwing exception, then we are basically doing very similar to existing PGProperty.getInt() method, except it will be a float in return instead. Do you want to add a new getFloat method in PGProperty class for reuse in future? Or you like to just keep the changes in Driver.java for now?